save-core/src/commonMain/kotlin/org/cqfn/save/core/files/ConfigDetector.kt

```diff
@@ -18,57 +19,76 @@ class ConfigDetector {
      * @param file a [Path] from which SAVE config file should be built.
      * @return [TestConfig] or null if no suitable config file has been found.
      */
-    fun configFromFile(file: Path): TestConfig? = discoverConfigWithParents(file)
-        ?.also { config ->
-            // fill children for parent configs
-            config.parentConfigs(wihSelf = true).toList().reversed()
-                .zipWithNext().forEach { (parent, child) ->
-                    parent.childConfigs.add(child)
-                }
-            // discover all descendant configs of [config]
-            val locationsFlattened = config.directory.findAllFilesMatching { it.isSaveTomlConfig() }.flatten()
-            val configs = mutableListOf(config)
-            locationsFlattened
-                .drop(1)  // because [config] will be discovered too
-                .forEachIndexed { index, path ->
-                    val parentConfig = configs.find { discoveredConfig ->
-                        discoveredConfig.location ==
-                                locationsFlattened.take(index + 1).reversed().find { it.parent in path.parents() }!!
-                    }!!
-                    configs.add(
-                        TestConfig(
+    fun configFromFile(testConfig: String?): TestConfig {
+        // testConfig is validated in the beginning and cannot be null
```

Comment:
```suggestion
    requireNotNull(testConfig) { "Provided path ${saveProperties.testConfig} doesn't correspond to a valid save.toml file" }
```
Maybe, add `requireNotNull` here, like it was in `SAve.kt` before?

Reply:
@petertrr please check save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt